src/main/java/teammates/ui/webapi/action/GetFeedbackSessionsAction.java

```diff
@@ -20,35 +24,72 @@ protected AuthType getMinAuthLevel() {
 
     @Override
     public void checkSpecificAccessControl() {
+        String entityType = getNonNullRequestParamValue(Const.ParamsNames.ENTITY_TYPE);
+
+        if (!(entityType.equals(Const.EntityType.STUDENT) || entityType.equals(Const.EntityType.INSTRUCTOR))) {
+            throw new UnauthorizedAccessException("entity type not supported.");
+        }
+
         String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
-        if (courseId == null) {
-            gateKeeper.verifyInstructorPrivileges(logic.getAccount(userInfo.getId()));
+
+        if (entityType.equals(Const.EntityType.STUDENT)) {
+            if (!userInfo.isStudent) {
+                throw new UnauthorizedAccessException("User " + userInfo.getId()
+                        + " does not have student privileges");
+            }
+
+            if (courseId != null) {
+                CourseAttributes courseAttributes = logic.getCourse(courseId);
+                gateKeeper.verifyAccessible(logic.getStudentForGoogleId(courseId, userInfo.getId()), courseAttributes);
+            }
         } else {
-            CourseAttributes courseAttributes = logic.getCourse(courseId);
-            gateKeeper.verifyAccessible(logic.getInstructorForGoogleId(courseId, userInfo.getId()), courseAttributes);
+            if (!userInfo.isInstructor) {
+                throw new UnauthorizedAccessException("User " + userInfo.getId()
+                        + " does not have instructor privileges");
+            }
+
+            if (courseId != null) {
+                CourseAttributes courseAttributes = logic.getCourse(courseId);
+                gateKeeper.verifyAccessible(logic.getInstructorForGoogleId(courseId, userInfo.getId()), courseAttributes);
+            }
         }
     }
 
     @Override
     public ActionResult execute() {
         String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String entityType = getNonNullRequestParamValue(Const.ParamsNames.ENTITY_TYPE);
```

Comment:
There is at least one place in the frontend that use the API. I think you need to add entity type as a parameter for them. It would be better to abstract the method in service to reduce code duplication.